File: backend/app/schemas/imports.py
"""文档导入 HTTP API 数据契约。"""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field

from app.services.import_tasks import ImportTaskRecord, ImportTaskStatus


class ImportAcceptedResponse(BaseModel):
    message: str
    task_id: str
    status: ImportTaskStatus
    filename: str
    status_url: str


class ImportTaskErrorResponse(BaseModel):
    node: str | None = None
    message: str


class ImportTaskResponse(BaseModel):
    task_id: str
    filename: str
    status: ImportTaskStatus
    done_nodes: list[str] = Field(default_factory=list)
    running_node: str | None = None
    node_durations_ms: dict[str, float] = Field(default_factory=dict)
    chunk_count: int = Field(default=0, ge=0)
    item_name: str = ""
    milvus_collection_name: str = ""
    error: ImportTaskErrorResponse | None = None
    created_at: datetime
    updated_at: datetime

    @classmethod
    def from_record(cls, record: ImportTaskRecord) -> ImportTaskResponse:
        error = None
        if record.error_message is not None:
            error = ImportTaskErrorResponse(
                node=record.error_node,
                message=record.error_message,
            )
        return cls(
            task_id=record.task_id,
            filename=record.filename,
            status=record.status,
            done_nodes=list(record.done_nodes),
            running_node=record.running_node,
            node_durations_ms=record.node_durations_ms,
            chunk_count=record.chunk_count,
            item_name=record.item_name,
            milvus_collection_name=record.milvus_collection_name,
            error=error,
            created_at=record.created_at,
            updated_at=record.updated_at,
        )

File: frontend/src/types/imports.ts
export type BackendImportStatus = "queued" | "processing" | "completed" | "failed"
export type ImportViewStatus = "uploading" | BackendImportStatus | "connection_error"
export type ImportFileKind = "pdf" | "markdown"

export interface ImportAcceptedResponse {
  message: string
  task_id: string
  status: BackendImportStatus
  filename: string
  status_url: string
}

export interface ImportTaskError {
  node: string | null
  message: string
}

export interface ImportTaskResponse {
  task_id: string
  filename: string
  status: BackendImportStatus
  done_nodes: string[]
  running_node: string | null
  node_durations_ms: Record<string, number>
  chunk_count: number
  item_name: string
  milvus_collection_name: string
  error: ImportTaskError | null
  created_at: string
  updated_at: string
}

export interface ImportTaskView {
  localId: string
  file: File
  kind: ImportFileKind
  status: ImportViewStatus
  taskId: string
  statusUrl: string
  doneNodes: string[]
  runningNode: string | null
  nodeDurationsMs: Record<string, number>
  chunkCount: number
  itemName: string
  collectionName: string
  errorMessage: string
  retryCount: number
}

export interface PipelineStep {
  id: string
  label: string
  shortLabel: string
}

export interface ApiErrorPayload {
  error?: {
    code?: string
    message?: string
  }
}
